Code maintenance and preparation for Run III in the DQMOffline ZCounting

ThOliveira23 opened the pull request:
**PR description:**

At the python configuration file there are some modifications in the values for pT, mass range and PU and the change of the trigger (from HLT_IsoMu_27 to HLT_IsoMu_24), concerning the preparation for Run3.


At the plugins, the definition of a Custom Tight ID for the muon, with no consideration of the primary vertices quantities. Also a bug fixing. 

**PR validation:**

Compilation of the cmssw.
A test using the DQMOffline/Lumi/test/ZCounting_cfg.py to produce the root file.

**if this PR is a backport please specify the original PR and why you need to backport that PR:**

There isn’t a backport intended.

ThOliveira23 committed:
First test

ThOliveira23 committed:
Merge pull request #2 from ThOliveira23/from-CMSSW_12_4_0_pre4

First test

ThOliveira23 committed:
CustomTightID header declaration

ThOliveira23 committed:
CustomTightID header declaration

ThOliveira23 committed:
Bug fixing at ZCounting.cc

ThOliveira23 committed:
Accept jets without PV

ThOliveira23 committed:
CustomTightID function

ThOliveira23 committed:
Config file for early Run3

ThOliveira23 committed:
Merge pull request #3 from ThOliveira23/from-CMSSW_12_4_0_pre4

From cmssw 12 4 0 pre4

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30051

- This PR adds an extra 24KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30051/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30051/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

ThOliveira23 committed:
Code format building

ThOliveira23 committed:
Merge pull request #4 from ThOliveira23/from-CMSSW_12_4_0_pre4

Code format building

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30074

- This PR adds an extra 24KB to repository

cmsbuild commented:
A new Pull Request was created by  @ThOliveira23 (Thales Oliveira) for master.

It involves the following packages:

- DQMOffline/Lumi (**dqm**)


@emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @micsucmed, @rvenditti can you please review it and eventually sign? Thanks.
@rociovilar this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

ThOliveira23 commented:
@cmsbuild Please test

emanueleusai commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-eafb4e/24844/summary.html
**COMMIT**: 7dfc23a123132082dcc900f7cf8c1b68ac872e65
**CMSSW**: CMSSW_12_5_X_2022-05-18-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/37981/24844/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3672423
 - DQMHistoTests: Total failures: 13
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3672387
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 49 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 208 log files, 45 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-05-18-2300+eafb4e/50392/triggerResults): no differences found

emanueleusai commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

emanueleusai commented on DQMOffline/Lumi/plugins/ZCounting.cc:
```diff
@@ -609,8 +609,8 @@ void ZCounting::analyzeElectrons(const edm::Event& iEvent, const edm::EventSetup
   // Good vertex requirement
-  if (nvtx == 0)
-    return;
+  //if (nvtx == 0)
+  //  return;
```
can you please remove these two comment lines?

ThOliveira23 replied:
The two lines were been removed.

ThOliveira23 committed:
Removal of the vertex lines

ThOliveira23 committed:
Merge pull request #5 from ThOliveira23/from-CMSSW_12_4_0_pre4

Removal of the vertex lines

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30112

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #37981 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @micsucmed, @rvenditti can you please check and sign again.

perrotta commented on DQMOffline/Lumi/plugins/ZCounting.cc:
```diff
@@ -609,8 +609,6 @@ void ZCounting::analyzeElectrons(const edm::Event& iEvent, const edm::EventSetup
   }
 
   // Good vertex requirement
```
Now this comment hangs uselessly... I would also remove it, together with the two other lines
Once you do so, could you please squash all commits into one?

perrotta replied:
The outdated comment is still there: could you please remove it (and squash all commits after it)?

ThOliveira23 replied:
I would like to maintain the vertex condition for the electrons, since it was supposed to be removed only for the muons (analyzeMuons).

ThOliveira23 committed:
Merge branch 'cms-sw:master' into master

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30133

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #37981 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @micsucmed, @rvenditti can you please check and sign again.

ThOliveira23 committed:
Code maintenance for Run III in the DQMOffline ZCounting

ThOliveira23 committed:
Merge branch 'master' into from-CMSSW_12_5_0_pre1_v2

ThOliveira23 committed:
Merge pull request #6 from ThOliveira23/from-CMSSW_12_5_0_pre1_v2

Code maintenance for Run III in the DQMOffline ZCounting

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30137

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #37981 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @micsucmed, @rvenditti can you please check and sign again.

emanueleusai commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-eafb4e/24974/summary.html
**COMMIT**: ac9cae3d2b5bc3c89211af9c49931815bba1a989
**CMSSW**: CMSSW_12_5_X_2022-05-24-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37981/24974/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3650985
 - DQMHistoTests: Total failures: 8
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3650955
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 208 log files, 45 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-05-24-2300+eafb4e/50493/triggerResults): no differences found

emanueleusai commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented on DQMOffline/Lumi/plugins/ZCounting.cc:
```diff
@@ -864,6 +858,20 @@ bool ZCounting::isMuonTriggerObj(const ZCountingTrigger::TTrigger& triggerMenu,
+
+  bool muID = isGoodMuon(muon, muon::GlobalMuonPromptTight) && (muon.numberOfMatchedStations() > 1);
+
+  bool hits = muon.innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5 &&
+              muon.innerTrack()->hitPattern().numberOfValidPixelHits() > 0;
+
+  return muID && hits;
```
```suggestion
  bool muIdAndHits = muId &&
              muon.innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5 &&
              muon.innerTrack()->hitPattern().numberOfValidPixelHits() > 0;

  return muIdAndHits;
```
can save some computation if muID is already `false`

ThOliveira23 replied:
After the modifications, how can I make the squash of all the commits?

perrotta replied:
> After the modifications, how can I make the squash of all the commits?

After made all modifications, and committed, you can:
```
git log
git reset --soft [commit]    # where [commit] is the last commit from the log previous to the first one of yours
git commit -m "any comment"
git push -f my-cmssw [theBranchName]
```

Now I see that you have several merge commits in the list of your commits... That could be a problem (see "Use of CMS commands" in http://cms-sw.github.io/tutorial-resolve-conflicts.html). Then, either forget my request to squash all commits into one, or be prepared to make a new PR from scratch by copying in another commit the modifications you made .

perrotta commented on DQMOffline/Lumi/python/ZCounting_cff.py:
```diff
@@ -4,7 +4,7 @@
                                  TriggerEvent    = cms.InputTag('hltTriggerSummaryAOD','','HLT'),
                                  TriggerResults  = cms.InputTag('TriggerResults','','HLT'),
-				 edmPVName       = cms.untracked.string('offlinePrimaryVertices'),
+				                         edmPVName       = cms.untracked.string('offlinePrimaryVertices'),
```
please re-indent this line

perrotta commented:
As far as I can see now, you can remove 
https://github.com/cms-sw/cmssw/blob/ac9cae3d2b5bc3c89211af9c49931815bba1a989/DQMOffline/Lumi/plugins/ZCounting.cc#L589-L608
as a whole, since you don't use `nvtx` any more for electrons

ThOliveira23 committed:
Update ZCounting.cc

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30203

- This PR adds an extra 32KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30203/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30203/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

ThOliveira23 committed:
Update ZCounting_cff.py

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30204

- This PR adds an extra 32KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30204/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30204/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

ThOliveira23 committed:
Code maintenance for Run III in the DQMOffline ZCounting.

ThOliveira23 committed:
Merge branch 'master' into from-CMSSW_12_5_0_pre1

ThOliveira23 committed:
Merge pull request #7 from ThOliveira23/from-CMSSW_12_5_0_pre1

Code maintenance for Run III in the DQMOffline ZCounting.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30205

- This PR adds an extra 20KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30205/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30205/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

ThOliveira23 committed:
Update ZCounting.cc, code format (l.610 and l.875)

Code maintenance: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30205/code-format.patch

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30206

- This PR adds an extra 88KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30206/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30206/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

ThOliveira23 committed:
Update ZCounting.cc

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37981/30207

- This PR adds an extra 92KB to repository

cmsbuild commented:
Pull request #37981 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @micsucmed, @rvenditti can you please check and sign again.

emanueleusai commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-eafb4e/25033/summary.html
**COMMIT**: 928e1674154dab26eb0e48677d45430e639af1c8
**CMSSW**: CMSSW_12_5_X_2022-05-26-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37981/25033/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3648322
 - DQMHistoTests: Total failures: 8
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3648292
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 208 log files, 45 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-05-26-2300+eafb4e/50549/triggerResults): no differences found

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)
